actions/rules/action_state_to_metastate.go

```diff
@@ -0,0 +1,416 @@
+package rules
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-wit/application"
+	"github.com/fabric8-services/fabric8-wit/convert"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+)
+
+type ActionStateToMetaState struct {
+	Db     application.DB
+	Ctx    context.Context
+	UserID *uuid.UUID
+}
+
+// make sure the rule is implementing the interface.
+var _ Action = ActionStateToMetaState{}
+
+func (act ActionStateToMetaState) contains(s []uuid.UUID, e uuid.UUID) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func (act ActionStateToMetaState) removeElement(s []uuid.UUID, e uuid.UUID) []uuid.UUID {
+	for idx, a := range s {
+		if a == e {
+			s = append(s[:idx], s[idx+1:]...)
+			// we don't return here as there may be multiple copies of e in s.
+		}
+	}
+	return s
+}
+
+func (act ActionStateToMetaState) difference(old []uuid.UUID, new []uuid.UUID) ([]uuid.UUID, []uuid.UUID) {
```

Comment:
We already have a function similar to this that you should consider using instead: https://github.com/fabric8-services/fabric8-wit/blob/8b6c84858e16aafc922c89577c47965c50b88328/id/slice.go#L14

See this file for examples of the `Diff`: https://github.com/fabric8-services/fabric8-wit/blob/8b6c84858e16aafc922c89577c47965c50b88328/id/slice_blackbox_test.go#L13

Reply:
Considered.